src/platform/ESP32/ConnectivityManagerImpl.h

```diff
@@ -45,24 +53,32 @@ class PlatformManagerImpl;
  * Concrete implementation of the ConnectivityManager singleton object for the ESP32 platform.
  */
 class ConnectivityManagerImpl final : public ConnectivityManager,
-                                      public Internal::GenericConnectivityManagerImpl<ConnectivityManagerImpl>,
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
                                       public Internal::GenericConnectivityManagerImpl_WiFi<ConnectivityManagerImpl>,
+#else
+                                      public Internal::GenericConnectivityManagerImpl_NoWiFi<ConnectivityManagerImpl>,
+#endif
 #if CHIP_DEVICE_CONFIG_ENABLE_CHIPOBLE
                                       public Internal::GenericConnectivityManagerImpl_BLE<ConnectivityManagerImpl>,
 #else
                                       public Internal::GenericConnectivityManagerImpl_NoBLE<ConnectivityManagerImpl>,
 #endif
-                                      public Internal::GenericConnectivityManagerImpl_NoThread<ConnectivityManagerImpl>
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+                                      public Internal::GenericConnectivityManagerImpl_Thread<ConnectivityManagerImpl>,
+#else
+                                      public Internal::GenericConnectivityManagerImpl_NoThread<ConnectivityManagerImpl>,
+#endif
+                                      public Internal::GenericConnectivityManagerImpl<ConnectivityManagerImpl>
 {
 
     // Allow the ConnectivityManager interface class to delegate method calls to
     // the implementation methods provided by this class.
     friend class ConnectivityManager;
 
 private:
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
```

Comment:
could we use some form of subclassing instead of `#ifdef` to compile-time switch code?

I would find an organization of `ConnectivityManagerImpl_Wifi.h`and `ConnectivitymanagerImpl_Thread.h` and cpp much easier to follow and review.

Reply:
The name of `ConnectivityManagerImpl.h` couldn't be changed because `CONNECTIVITYMANAGERIMPL_HEADER` is defined in [here](https://github.com/project-chip/connectedhomeip/blob/master/src/include/platform/ConnectivityManager.h#L267).  So I add  `ConnectivityManagerImpl_WiFi.h`, `ConnectivitymanagerImpl_Thread.h` and cpp files and modify `ConnectivityManagerImpl.h`:
```
#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
#include <platform/ESP32/ConnectivityManagerImpl_WiFi.h>
#elif CHIP_DEVICE_CONFIG_ENABLE_THREAD
#include <platform/ESP32/ConnectivityManagerImpl_Thread.h>
#endif
```